generators/generator-base.js

```diff
@@ -1108,22 +1108,9 @@ module.exports = class extends PrivateBase {
      * @param {any} options - options to pass
      */
     composeExternalModule(npmPackageName, subGen, options) {
-        let generatorTocall = path.join(process.cwd(), 'node_modules', npmPackageName, 'generators', subGen);
-        try {
-            if (!fs.existsSync(generatorTocall)) {
-                this.debug('using global module as local version could not be found in node_modules');
-                generatorTocall = path.join(npmPackageName, 'generators', subGen);
-            }
-            this.debug('Running yeoman compose with options: ', generatorTocall, options);
-            this.composeWith(require.resolve(generatorTocall), options);
-        } catch (err) {
-            this.debug('ERROR:', err);
-            const generatorName = npmPackageName.replace('generator-', '');
-            const generatorCallback = `${generatorName}:${subGen}`;
-            // Fallback for legacy modules
-            this.debug('Running yeoman legacy compose with options: ', generatorCallback, options);
-            this.composeWith(generatorCallback, options);
-        }
+        const generatorName = npmPackageName.replace('generator-', '');
+        const generatorCallback = `${generatorName}:${subGen}`;
+        this.composeWith(generatorCallback, options);
```

Comment:
So this now always uses the legacy wy of calling something in compose right. I remember the docs saying to use the `this.composeWith(require.resolve(generatorTocall), options)` way

Reply:
I was not aware of that.
https://yeoman.io/authoring/composability.html indeed says `Even though it is not an encouraged practice, you can also pass a generator namespace to composeWith.`
But the jsdoc doesn't warn about that particular usage: https://github.com/yeoman/generator/blob/7b05f030622e05c9fa62b482f8ebe4a706c469ba/lib/index.js#L512

The goal is indeed to avoid `require.resolve()` that doesn't always work.

Reply:
composeExternalModule is used by external modules, removing 'generator-' from modules packageName probably will break modules support.
Moving this new logic to composeBlueprint and deprecating modules support (callHooks, composeExternalModule) seems the way to go.

Reply:
I don't understand how it would break modules support. 
Removing 'generator-' is just the way to transform a npm package name to a Yeoman namespace. There's no differences between blueprints and modules in that perspective.

And I'm against deprecating modules support, they have their value, and always using blueprint is like using a sledgehammer to crack a nut.

Reply:
Does modules without 'generator-' prefix works?

Reply:
Modules are Yeoman generators, so they should use 'generator-' prefix. See Basic rules for a JHipster module in https://www.jhipster.tech/modules/creating-a-module/